### ISSUE TITLE
v3.1.0@e25f68a compile issue crc4.S -fPIC flag

### ISSUE THREAD

**Issue by baslr** [arangodb/arangodb#2147]
```
# CFLAGS="-fPIC" CXXFLAGS="-fPIC" cmake -DVERBOSE=true -DCMAKE_BUILD_TYPE=Release ..
...
-- Compiler type GNU: /usr/bin/c++
-- CMAKE_TARGET_ARCHITECTURES:x86_64
-- CMAKE_HOST_SYSTEM_PROCESSOR:x86_64
-- OPT_momit_leaf_frame_pointer:-momit-leaf-frame-pointer
-- Info BASE_FLAGS:     -Wall -Wextra -Wno-unused-parameter -mno-sse2 -mno-sse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -mno-sse4a -mno-avx -mno-xop -mno-fma4 -mno-avx2  -W -Wextra -Wall -Wsign-compare -Wshadow -Wno-unused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer -fno-builtin-memcmp -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers  -DROCKSDB_MALLOC_USABLE_SIZE -isystem -fPIC -DOS_LINUX -DROCKSDB_PLATFORM_POSIX -DROCKSDB_LIB_IO_POSIX
-- Info BASE_C_FLAGS:   -fPIC    -fPIC -std=gnu89
-- Info BASE_CXX_FLAGS: -fPIC  -fPIC -std=c++11
-- Info BASE_LD_FLAGS:
-- Info BASE_LIBS:
--
-- Info CMAKE_C_FLAGS_RELEASE:   -Wall -Wextra -Wno-unused-parameter -mno-sse2 -mno-sse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -mno-sse4a -mno-avx -mno-xop -mno-fma4 -mno-avx2  -W -Wextra -Wall -Wsign-compare -Wshadow -Wno-unused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer -fno-builtin-memcmp -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers  -DROCKSDB_MALLOC_USABLE_SIZE -isystem -fPIC -DOS_LINUX -DROCKSDB_PLATFORM_POSIX -DROCKSDB_LIB_IO_POSIX -fPIC    -fPIC -std=gnu89 -O3 -fomit-frame-pointer -DNDEBUG
-- Info CMAKE_CXX_FLAGS_RELEASE: -Wall -Wextra -Wno-unused-parameter -mno-sse2 -mno-sse3 -mno-ssse3 -mno-sse4.1 -mno-sse4.2 -mno-sse4a -mno-avx -mno-xop -mno-fma4 -mno-avx2  -W -Wextra -Wall -Wsign-compare -Wshadow -Wno-unused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer -fno-builtin-memcmp -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers  -DROCKSDB_MALLOC_USABLE_SIZE -isystem -fPIC -DOS_LINUX -DROCKSDB_PLATFORM_POSIX -DROCKSDB_LIB_IO_POSIX -fPIC  -fPIC -std=c++11 -O3 -fomit-frame-pointer -DNDEBUG
-- Info CMAKE_EXE_LINKER_FLAGS:
--
-- JNI library is disabled
-- -- Will compile in hand-optimized assembler code for CRC32.
-- building for git revision: tags/v3.1.0-0-ge25f68a
-- Configuring done
-- Generating done
-- Build files have been written to: /root/arangodb/build

make -j 1
...
[ 53%] Building ASM object lib/CMakeFiles/arango.dir/Basics/crc4.S.o
...
Scanning dependencies of target arangorestore
[ 65%] Building CXX object arangosh/CMakeFiles/arangorestore.dir/__/lib/Basics/WorkMonitorDummy.cpp.o
[ 65%] Building CXX object arangosh/CMakeFiles/arangorestore.dir/Restore/RestoreFeature.cpp.o
[ 65%] Building CXX object arangosh/CMakeFiles/arangorestore.dir/Restore/arangorestore.cpp.o
[ 65%] Building CXX object arangosh/CMakeFiles/arangorestore.dir/Shell/ClientFeature.cpp.o
[ 65%] Building CXX object arangosh/CMakeFiles/arangorestore.dir/Shell/ConsoleFeature.cpp.o
[ 65%] Building CXX object arangosh/CMakeFiles/arangorestore.dir/V8Client/ArangoClientHelper.cpp.o
[ 66%] Linking CXX executable ../bin/arangorestore
/usr/bin/ld: ../lib/libarango.a(crc4.S.o): relocation R_X86_64_32S against `.text' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Nonrepresentable section on output
collect2: error: ld returned 1 exit status
arangosh/CMakeFiles/arangorestore.dir/build.make:231: recipe for target 'bin/arangorestore' failed
make[2]: *** [bin/arangorestore] Error 1
CMakeFiles/Makefile2:1175: recipe for target 'arangosh/CMakeFiles/arangorestore.dir/all' failed
make[1]: *** [arangosh/CMakeFiles/arangorestore.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

Adding `-fPIC` to the env c / c++ flags has no effect.
Temporary workaround: cmake with `-DASM_OPTIMIZATIONS=0`

System is latest Debian Sid.

**Comment by baslr** [comment]
Probably related to gcc 6.2.

**Comment by dothebart** [comment]
hm, I installed a SID chroot, and currently fail to get to that point due to some openssl compile issues...

**Comment by baslr** [comment]
There is `libssl-dev` (openssl v1.1) and `libssl1.0-dev` (openssl v1.0)

**Comment by dothebart** [comment]
Stretch also has gcc 6.2; I also get:
```
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/6/../../../x86_64-linux-gnu/libjemalloc.a(arena.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
```
so probably need to recompile jemalloc first...

**Comment by baslr** [comment]
I don't use jemalloc. So don't have that problem ;)

**Comment by kvahed** [comment]
Fixed in 1f10de1
